common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -316,41 +363,33 @@ def destroy(self, request, uuid):
             )
 
         if is_refund and entitlement.is_entitlement_refundable():
-            with transaction.atomic():
-                # Revoke and refund the entitlement
-                if entitlement.enrollment_course_run is not None:
-                    self._unenroll_entitlement(
-                        entitlement=entitlement,
-                        course_run_key=entitlement.enrollment_course_run.course_id,
-                        user=request.user
-                    )
-
-                # Revoke the Course Entitlement and issue Refund
-                log.info(
-                    'Entitlement Refund requested for Course Entitlement[%s]',
-                    str(entitlement.uuid)
-                )
+            # Revoke the Course Entitlement and issue Refund
+            log.info(
+                'Entitlement Refund requested for Course Entitlement[%s]',
+                entitlement.uuid
+            )
 
-                REFUND_ENTITLEMENT.send(sender=None, course_entitlement=entitlement)
-                entitlement.expired_at_datetime = timezone.now()
-                entitlement.save()
+            try:
+                _process_revoke_and_unenroll_entitlement(course_entitlement=entitlement, is_refund=True)
+            except IntegrityError:
+                # This state is reached when there was a failure in revoke and refund process resulting
+                # in a reversion of DB changes
+                return Response(
+                    status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                    data={
+                        'message': 'Entitlement revoke and refund failed due to refund internal process failure'
+                    })
 
-                log.info(
-                    'Set expired_at to [%s] for course entitlement [%s]',
-                    entitlement.expired_at,
-                    entitlement.uuid
-                )
         elif not is_refund:
             if entitlement.enrollment_course_run is not None:
-                self._unenroll_entitlement(
-                    entitlement=entitlement,
+                _unenroll_entitlement(
+                    course_entitlement=entitlement,
                     course_run_key=entitlement.enrollment_course_run.course_id,
```

Comment:
Sanity check: calling this method with a trailing comma won't cause any issues (my 2.7 interpreter seems fine with it)?

Reply:
Please disregard, I see this is fine in python.

Reply:
I believe it might even be preferred style, but I would need to double check the Code Style guide.